nexus/db-queries/src/db/datastore/network_interface.rs

```diff
@@ -890,7 +901,16 @@ impl DataStore {
         // instance network interfaces, which require ListChildren on the
         // instance to list). As a logical proxy, we check for listing children
         // of the service IP pool.
-        let (authz_pool, _pool) = self.ip_pools_service_lookup(opctx).await?;
+        //
+        // TODO-cleanup: https://github.com/oxidecomputer/omicron/issues/8873.
+        // This authz check looks at the builtin services IP Pool, but we're
+        // listing _instance_ NICs. That seems to be authorizing against the
+        // wrong resource.
+        //
+        // But assuming this check is correct, both service pools have the same
+        // permissions, so the actual IP version here doesn't matter.
+        let (authz_pool, _pool) =
+            self.ip_pools_service_lookup(opctx, IpVersion::V4).await?;
```

Comment:
This is listing _all_ network interfaces, not just for a particular instance. Hence the comment. It is weird, and even weird is that the only call site, `instance_network_interfaces_all_list_batched`, itself has no call sites. So maybe this can just be deleted?

Reply:
Yeah -- it's actually used today, in tests. I thought about putting a `cfg(test)` on it, but I'd rather just handle it in a dedicated issue.